git/git.go

```diff
@@ -103,3 +103,33 @@ func getRemoteUrl(remoteName string) (string, error) {
 	}
 	return string(out), nil
 }
+
+func commandOutputOrDefault(cmd *exec.Cmd, defaultValue string) string {
+	output, err := cmd.CombinedOutput()
+
+	if err != nil {
+		return defaultValue
+	}
+
+	return strings.TrimSpace(string(output))
+}
+
+func Branch() string {
+	// Git 2.22 (Q2 2019)added `git branch --show-current`, but using
+	// `git rev-parse` works in all versions.
+	return commandOutputOrDefault(
+		exec.Command("git", "rev-parse", "--abbrev-ref", "HEAD"),
```

Comment:
If the user has a detached HEAD, this will return `HEAD`, which doesn't seem right, but it seems like some inaccuracy in these values is probably fine.